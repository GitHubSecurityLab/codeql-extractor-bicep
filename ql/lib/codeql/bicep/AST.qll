import ast.AstNodes
import ast.Literals
import ast.Conditionals
import ast.Calls
import ast.Expr
import ast.Stmts
import ast.Loops
import ast.Misc
import ast.Idents
import ast.Variables
import ast.Resources
import ast.Types
