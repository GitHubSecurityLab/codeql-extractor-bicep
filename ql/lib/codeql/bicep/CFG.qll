/** Provides classes representing the control flow graph. */

import codeql.Locations
import controlflow.BasicBlocks
import controlflow.ControlFlowGraph
import controlflow.CfgNodes as CfgNodes