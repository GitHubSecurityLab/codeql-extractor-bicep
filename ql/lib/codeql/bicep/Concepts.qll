private import codeql.bicep.AST
private import codeql.bicep.CFG

/**
 *  A Public Resource is a resource that is publicly accessible to the Internet.
 */
abstract class PublicResource extends Resource {
  /**
   *  Returns the property that indicates public access.
   */
  abstract Expr getPublicAccessProperty();
}

module Cryptography {
  abstract class WeakTlsVersion extends Resource {
    abstract Expr getWeakTlsVersionProperty();

    /**
     *  Returns true if the resource has a weak TLS version.
     * 
     *  1.0 and 1.1 are considered weak TLS versions.
     */
    predicate hasWeakTlsVersion() {
      exists(StringLiteral literal |
        literal = this.getWeakTlsVersionProperty() and
        literal.getValue().regexpMatch("^(1\\.0|1\\.1)$")
      )
    }
  }
}
