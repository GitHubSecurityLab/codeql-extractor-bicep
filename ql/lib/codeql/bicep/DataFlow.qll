import dataflow.Ssa
import dataflow.DataFlow
