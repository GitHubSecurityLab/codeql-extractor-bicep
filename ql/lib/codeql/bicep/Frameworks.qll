import frameworks.Microsoft.Compute
import frameworks.Microsoft.Network
import frameworks.Microsoft.Storage
import frameworks.Microsoft.Databases
import frameworks.Microsoft.KeyVault