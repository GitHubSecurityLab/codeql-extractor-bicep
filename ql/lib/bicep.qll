import codeql.Locations
import codeql.files.FileSystem
// AST
import codeql.bicep.AST
// CFG
import codeql.bicep.CFG
// Frameworks
import codeql.bicep.Frameworks
import codeql.bicep.Concepts
