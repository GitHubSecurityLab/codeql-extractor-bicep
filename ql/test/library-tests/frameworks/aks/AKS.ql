import bicep

query predicate mcr(AKS::ManagedContainerResource mcr) { any() }

query predicate mcrAgents(
  AKS::ManagedContainerResource mcr, AKS::ManagedContainerProperties::AgentPoolProfile agents
) {
  mcr.getAgentPoolProfiles() = agents
}

query predicate mcrOsProfile(AKS::ManagedContainerResource mcr, OsProfiles::OsProfile osProfile) {
  mcr.getOsProfile() = osProfile
}

query predicate mcrNetworkProfiles(
  AKS::ManagedContainerResource mcr, NetworkingProfiles::NetworkProfile networkProfile
) {
  mcr.getNetworkProfile() = networkProfile
}
