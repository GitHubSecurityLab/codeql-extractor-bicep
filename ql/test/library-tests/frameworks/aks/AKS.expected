mcr
| app.bicep:2:1:70:1 | ManagedContainerResource |
| example1-basic-cluster.bicep:5:1:43:1 | ManagedContainerResource |
| example2-production-cluster.bicep:5:1:188:1 | ManagedContainerResource |
| example3-ml-optimized-cluster.bicep:5:1:196:1 | ManagedContainerResource |
mcrAgents
| app.bicep:2:1:70:1 | ManagedContainerResource | app.bicep:9:7:18:7 | AgentPoolProfile |
| app.bicep:2:1:70:1 | ManagedContainerResource | app.bicep:19:7:26:7 | AgentPoolProfile |
| example1-basic-cluster.bicep:5:1:43:1 | ManagedContainerResource | example1-basic-cluster.bicep:12:7:19:7 | AgentPoolProfile |
| example2-production-cluster.bicep:5:1:188:1 | ManagedContainerResource | example2-production-cluster.bicep:27:7:46:7 | AgentPoolProfile |
| example2-production-cluster.bicep:5:1:188:1 | ManagedContainerResource | example2-production-cluster.bicep:47:7:67:7 | AgentPoolProfile |
| example2-production-cluster.bicep:5:1:188:1 | ManagedContainerResource | example2-production-cluster.bicep:68:7:91:7 | AgentPoolProfile |
| example3-ml-optimized-cluster.bicep:5:1:196:1 | ManagedContainerResource | example3-ml-optimized-cluster.bicep:22:7:36:7 | AgentPoolProfile |
| example3-ml-optimized-cluster.bicep:5:1:196:1 | ManagedContainerResource | example3-ml-optimized-cluster.bicep:37:7:55:7 | AgentPoolProfile |
| example3-ml-optimized-cluster.bicep:5:1:196:1 | ManagedContainerResource | example3-ml-optimized-cluster.bicep:56:7:79:7 | AgentPoolProfile |
| example3-ml-optimized-cluster.bicep:5:1:196:1 | ManagedContainerResource | example3-ml-optimized-cluster.bicep:80:7:96:7 | AgentPoolProfile |
mcrOsProfile
| app.bicep:2:1:70:1 | ManagedContainerResource | app.bicep:51:19:60:5 | OsProfile[linux] |
| example2-production-cluster.bicep:5:1:188:1 | ManagedContainerResource | example2-production-cluster.bicep:163:19:172:5 | OsProfile[linux] |
| example2-production-cluster.bicep:5:1:188:1 | ManagedContainerResource | example2-production-cluster.bicep:174:21:178:5 | OsProfile[windows] |
| example3-ml-optimized-cluster.bicep:5:1:196:1 | ManagedContainerResource | example3-ml-optimized-cluster.bicep:162:19:171:5 | OsProfile[linux] |
mcrNetworkProfiles
| app.bicep:2:1:70:1 | ManagedContainerResource | app.bicep:28:21:32:5 | NetworkProfile |
| example1-basic-cluster.bicep:5:1:43:1 | ManagedContainerResource | example1-basic-cluster.bicep:21:21:24:5 | NetworkProfile |
| example2-production-cluster.bicep:5:1:188:1 | ManagedContainerResource | example2-production-cluster.bicep:95:21:103:5 | NetworkProfile |
| example3-ml-optimized-cluster.bicep:5:1:196:1 | ManagedContainerResource | example3-ml-optimized-cluster.bicep:100:21:105:5 | NetworkProfile |
